t/re-split.t

```diff
@@ -1245,3 +1245,35 @@ GET /re
 qr/\[TRACE   \d+/
 --- no_error_log
 [error]
+
+
+
+=== TEST 38: remaining characters are matched by regex
```

Comment:
It'd be more maintainable if we split the 2 test cases (with and without max) in two separate tests - one could fail without the other one failing, this helps narrowing down the issue for a maintainer tackling it.